hardware_interface/src/resource_manager.cpp

```diff
@@ -25,16 +25,46 @@
 #include "hardware_interface/actuator.hpp"
 #include "hardware_interface/actuator_interface.hpp"
 #include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_component_info.hpp"
 #include "hardware_interface/sensor.hpp"
 #include "hardware_interface/sensor_interface.hpp"
 #include "hardware_interface/system.hpp"
 #include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
 #include "lifecycle_msgs/msg/state.hpp"
 #include "pluginlib/class_loader.hpp"
 #include "rcutils/logging_macros.h"
 
+#define COMPONENT_NAME_COMPARE [&](const auto & name) { return name == component.get_name(); }
+#define COMPONENT_NAME_COMPARE_INV \
+  [&](const auto & component) { return component.get_name() == component_name; }
+
 namespace hardware_interface
 {
+auto trigger_hardware_state_transition =
```

Comment:
this doesn't trigger anything, just gets whether we have the right state and adds some prints on top.
how about a simple helper function (not an lambda) called something ilke `check_and_print_hardware_state()`?

Reply:
but... we also have the same logic down below, like 
```
bool result = hardware.initialize(hardware_info).id() ==
                  lifecycle_msgs::msg::State::PRIMARY_STATE_UNCONFIGURED;
``` 
followed by essentially the same prints

Reply:
It actually triggers transition, but it was a bit too hidden. I have adjusted the code now to be clearer.

Reply:
thanks